### PR TITLE
Updated multiple class and states with Javadoc comments for Chess Game

### PR DESCRIPTION
**Pull request by ErdunE**
### ChessDriver.java
* Entry point for the game, managing game initialization.	
* Add JavaDoc comment
### ChessGame.java
* Add start function to hand game flow and state transitions. 
* Initializes the game and manages the sequence of player actions.
* Add script to test the chess game automatically
* Add JavaDoc comment
### Board.java
* Sets up the chessboard with initial pieces and provides methods for piece placement and move validation.
### Node.java
* Supports board structure with node-based positions for each piece.
### Logic.java
* Provides move logic for different pieces.
### GameStartState.java
* Manages the initial game setup, transitioning to the NormalPlayState.
### CheckmateState.java
* Handles the endgame scenario where a player is in checkmate.

### CODE REVIEW

File: ChessStatePattern/src/states/CheckmateState.java [hunk 1]
```diff
@@ -3,21 +3,62 @@
...
+    /**
+     * Displays the end-of-game message indicating that the game has concluded with checkmate.
+     * This method also announces the winner based on the current player in the ChessGame instance.
+     * Since this is the terminal state, no further state transitions occur.
+     *
+     * @return the current instance of CheckmateState, as no further transitions are allowed
+     */
+    @Override
+    public State checkMate() {
+        System.out.println("Game Over. Checkmate!");
```
Comment (xinyuhwang) [inline ChessStatePattern/src/states/CheckmateState.java]: Maybe switch the order of these two? Checkmate! Game Over.

File: ChessStatePattern/src/pieces/Logic.java [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
  */
 
 public class Logic {
+    private State currentState;
+
+    public Logic() {
+        this.currentState = new GameStartState(new ChessGame());
+    }
+
+    public void executeMove() {
```
Comment (xinyuhwang) [inline ChessStatePattern/src/pieces/Logic.java:24]: Java doc comments?

File: ChessStatePattern/src/pieces/Logic.java [hunk 3]
```diff
@@ -1,11 +1,31 @@
...
+        currentState = currentState.normalPlay();
+    }
 
-}
+    public void checkForCheck() {
```
Comment (xinyuhwang) [inline ChessStatePattern/src/pieces/Logic.java:28]: The naming seems a little confusing. Maybe call it isCheck? 
Java doc comments?

File: ChessStatePattern/src/states/PlayerTurnSwitchState.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package states;
+
+import main.ChessGame;
+
```
Comment (xinyuhwang) [inline ChessStatePattern/src/states/PlayerTurnSwitchState.java:4]: Java doc comments?

**Review by kimblewick — DISMISSED** [review]

File: ChessStatePattern/src/pieces/Logic.java (hunk 2 above)
Comment (ErdunE) [reply ChessStatePattern/src/pieces/Logic.java:24]: We didn't implement it at this time.

File: ChessStatePattern/src/states/PlayerTurnSwitchState.java (hunk 4 above)
Comment (ErdunE) [reply ChessStatePattern/src/states/PlayerTurnSwitchState.java:4]: This one will be covered by Raj

File: ChessStatePattern/src/pieces/Logic.java (hunk 3 above)
Comment (ErdunE) [reply ChessStatePattern/src/pieces/Logic.java:28]: Changed, but we won't implement it.

**Review by kimblewick — DISMISSED** [review]

**Review by xinyuhwang — DISMISSED** [review]

**Review by arete7-kai — DISMISSED** [review]

**Review by kimblewick — APPROVED** [review]

**Review by xinyuhwang — APPROVED** [review]

**Pull request merged**